test/unit/instance-id/instance-id-request.spec.ts

```diff
@@ -0,0 +1,133 @@
+/*!
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as _ from 'lodash';
+import {expect} from 'chai';
+import * as chai from 'chai';
+import * as nock from 'nock';
+import * as sinon from 'sinon';
+import * as sinonChai from 'sinon-chai';
+import * as chaiAsPromised from 'chai-as-promised';
+
+import * as utils from '../utils';
+import * as mocks from '../../resources/mocks';
+
+import {FirebaseApp} from '../../../src/firebase-app';
+import {HttpRequestHandler} from '../../../src/utils/api-request';
+import { FirebaseInstanceIdRequestHandler } from '../../../src/instance-id/instance-id-request';
+
+chai.should();
+chai.use(sinonChai);
+chai.use(chaiAsPromised);
+
+describe('FirebaseInstanceIdRequestHandler', () => {
+  let mockApp: FirebaseApp;
+  let projectId: string = 'test-project-id';
+  let mockedRequests: nock.Scope[] = [];
+  let stubs: sinon.SinonStub[] = [];
+  let mockAccessToken: string = utils.generateRandomAccessToken();
+  let expectedHeaders: Object;
+
+  before(() => utils.mockFetchAccessTokenRequests(mockAccessToken));
+
+  after(() => {
+    stubs = [];
+    nock.cleanAll();
+  });
+
+  beforeEach(() => {
+    mockApp = mocks.app();
+    expectedHeaders = {
+      Authorization: 'Bearer ' + mockAccessToken,
+    };
+  });
+
+  afterEach(() => {
+    _.forEach(stubs, (stub) => stub.restore());
+    _.forEach(mockedRequests, (mockedRequest) => mockedRequest.done());
+    return mockApp.delete();
+  });
+
+  describe('Constructor', () => {
+    it('should succeed with a FirebaseApp instance', () => {
+      expect(() => {
+        return new FirebaseInstanceIdRequestHandler(mockApp, projectId);
+      }).not.to.throw(Error);
+    });
+  });
+
+  describe('deleteInstanceId', () => {
```

Comment:
Looks good. One suggestion: i think it would be a good idea to also confirm the error code too.

Reply:
Done